akka-javasdk/src/main/scala/akka/javasdk/impl/GrpcEndpointDescriptorFactory.scala

```diff
@@ -20,14 +22,22 @@ import akka.runtime.sdk.spi.ComponentOptions
 import akka.runtime.sdk.spi.GrpcEndpointDescriptor
 import akka.runtime.sdk.spi.GrpcEndpointRequestConstructionContext
 import akka.runtime.sdk.spi.MethodOptions
+import io.grpc.Status
 import io.opentelemetry.api.trace.Span
 
+import java.util.concurrent.CompletionException
 import scala.concurrent.Future
 
 object GrpcEndpointDescriptorFactory {
 
   val logger: org.slf4j.Logger = org.slf4j.LoggerFactory.getLogger(GrpcEndpointDescriptorFactory.getClass)
 
+  private def failedCompletionMapper(system: ClassicActorSystem): PartialFunction[Throwable, Trailers] = {
+    case e: CompletionException =>
+      if (e.getCause == null) Trailers(Status.INTERNAL)
+      else GrpcExceptionHandler.defaultMapper(system.classicSystem)(e.getCause)
+  }
```

Comment:
Would be better to compose all in one place (here) and just use that as param, than create two different default mappers in different locations in the soruces and orElse later.

Reply:
Thanks, see now.